src/Avalonia.Controls/Calendar/CalendarBlackoutDatesCollection.cs

```diff
@@ -11,26 +12,15 @@
 
 namespace Avalonia.Controls.Primitives
 {
-    public sealed class CalendarBlackoutDatesCollection : ObservableCollection<CalendarDateRange>
+    public class CalendarBlackoutDatesCollection : ObservableCollection<CalendarDateRange>, IAvaloniaReadOnlyList<CalendarDateRange>
```

Comment:
Why not just inherit from `AvaloniaList<>` here rather than `ObservableCollection<>` if you need an `IAvaloniaReadOnlyList` interface?

Reply:
@grokys - The inheritance story is much better for `ObservaleCollection<>` compared to `AvaloniaList<>`. Sub-classes of `AvaloniaList<>` have no way of processing items set through their indexer and require more overrides to handle other changes. I have some interest in improving `AvaloniaList<>` but it would likely involve breaking changes, I can create an issue to discuss this further. `IAvaloniaReadOnlyList <>` isn't strictly required, but `ObservaleCollection<>` already implements it and the `ForEachItem` extension method is really nice.

Reply:
Sure, please do open an issue to discuss that!